vendor/calcite-1_19_0/build.gradle

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins { id 'org.apache.beam.vendor-java' }
+
+description = "Apache Beam :: Vendored Dependencies :: Calcite 1.19.0"
+
+group = "org.apache.beam"
+version = "0.1"
+
+def calcite_version = "1.19.0"
+def avatica_version = "1.15.0"
+def prefix = "org.apache.beam.vendor.calcite.v1_19_0"
+
+vendorJava(
+  dependencies: [
+    "org.apache.calcite:calcite-core:$calcite_version",
+    "org.apache.calcite:calcite-linq4j:$calcite_version",
+    "org.apache.calcite.avatica:avatica-core:$avatica_version",
+  ],
+  relocations: [
+    "org.apache.calcite": "${prefix}.org.apache.calcite",
+
+    // Calcite has Guava on its API surface
+    "com.google.common": "org.apache.beam.vendor.guava.v26_0_jre.com.google.thirdparty",
+    "com.google.thirdparty": "org.apache.beam.vendor.guava.v26_0_jre.com.google.thirdparty",
+
+    // Making this self-contained, includes Calcite's dependencies
+    "org.apache.commons": "${prefix}.org.apache.beam",
+    "com.esri": "${prefix}.com.esri",
+    "com.fasterxml": "${prefix}.com.fasterxml",
+    "com.yahoo": "${prefix}.com.yahoo",
+    "com.jayway": "${prefix}.com.jayway",
+    "org.pentaho": "${prefix}.org.pentaho",
+    "org.codehaus": "${prefix}.org.codehaus",
+    "org.slf4j": "${prefix}.org.slf4j",
```

Comment:
Consider not shading slf4j to retain logging from Calcite and any generated code as well just declare slf4j-api as a dependency of this vendored artifact.